pkg/auth/auth.go

```diff
@@ -79,20 +80,20 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		if !opts.AcceptUnspecifiedRegistry {
 			return errors.New("please provide a registry to login to")
 		}
-		if server, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
+		if key, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
 			return err
 		}
-		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
-	} else {
-		server = getRegistryName(args[0])
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", key)
 	}
-	authConfig, err := config.GetCredentials(systemContext, server)
+
+	authConfig, domain, err := getCredentials(systemContext, key, opts.AcceptRepositories)
```

Comment:
Before, `server = getRegistryName(args[0])` would normalize the input; now it seems that `key` is intended to be passed directly to `SetCredentials` and the like. That seems wrong. Differentiate between `userInput` to the parser, and `key` and `registry` (`domain`? that’s imprecise, it can be a host:port).

Reply:
Yes, I agree, preserved the previous behavior.